### PR TITLE
Expose dashboard layout and statuses API

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

I cleaned up the format that was used by the API, migrated the existing endpoints to OCS (the LayoutApiController never had a route so it was not possible to update the layout or statuses before) and added endpoints to get the current layout and statuses.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dashboard/appinfo/routes.php [hunk 1]
```diff
@@ -28,12 +28,14 @@
...
-		['name' => 'dashboard#updateStatuses', 'url' => '/statuses', 'verb' => 'POST'],
 	],
 	'ocs' => [
 		['name' => 'dashboardApi#getWidgets', 'url' => '/api/v1/widgets', 'verb' => 'GET'],
 		['name' => 'dashboardApi#getWidgetItems', 'url' => '/api/v1/widget-items', 'verb' => 'GET'],
 		['name' => 'dashboardApi#getWidgetItemsV2', 'url' => '/api/v2/widget-items', 'verb' => 'GET'],
+		['name' => 'dashboardApi#getLayout', 'url' => '/layout', 'verb' => 'GET'],
+		['name' => 'dashboardApi#updateLayout', 'url' => '/layout', 'verb' => 'POST'],
+		['name' => 'dashboardApi#getStatuses', 'url' => '/statuses', 'verb' => 'GET'],
+		['name' => 'dashboardApi#updateStatuses', 'url' => '/statuses', 'verb' => 'POST'],
```
Comment (julien-nc) [inline apps/dashboard/appinfo/routes.php:39]: Don't you wanna use the new routing system here?

File: apps/dashboard/src/DashboardApp.vue [hunk 2]
```diff
@@ -394,13 +394,16 @@ export default {
 			}
 		},
 		enableStatus(app) {
-			this.enabledStatuses[app] = true
+			this.enabledStatuses.push(app);
```
Comment (julien-nc) [inline apps/dashboard/src/DashboardApp.vue:397]:
```suggestion
			this.enabledStatuses.push(app)
```

File: apps/dashboard/src/DashboardApp.vue [hunk 3]
```diff
@@ -394,13 +394,16 @@ export default {
...
 			this.saveStatuses()
 		},
 		disableStatus(app) {
-			this.enabledStatuses[app] = false
-			const i = this.registeredStatus.findIndex((s) => s === app)
+			let i = this.enabledStatuses.findIndex((s) => s === app)
+			if (i !== -1) {
+				this.enabledStatuses.splice(i, 1)
+			}
+			i = this.registeredStatus.findIndex((s) => s === app)
```
Comment (julien-nc) [inline apps/dashboard/src/DashboardApp.vue:406]:
```suggestion
			const i = this.enabledStatuses.findIndex((s) => s === app)
			if (i !== -1) {
				this.enabledStatuses.splice(i, 1)
			}
			const j = this.registeredStatus.findIndex((s) => s === app)
```
And adjust the rest of the method.
I had to read it twice to realize why i was not a const. Slightly clearer like this IMO. We can afford declaring one more integer.

**Review by julien-nc — CHANGES_REQUESTED** [review]
Looks good. A few minor change suggestions.

File: apps/dashboard/appinfo/routes.php (hunk 1 above)
Comment (provokateurin) [reply apps/dashboard/appinfo/routes.php:39]: Sure why not, this PR was written before it was in place.

**Review by julien-nc — APPROVED** [review]
:+1:

**Review by st3iny — APPROVED** [review]
Tested and works.

Nit: I would prefer the new routes to be `/api/v1/...` as they are the first versions of the respective APIs.

**Comment by provokateurin** [comment]
Yeah I first had them as v1, but I'm not sure what makes most sense here. To me it would feel a bit weird that older versions support v2 but not all of the updated v1, so I went with v3 instead :woman_shrugging:

**Comment by provokateurin** [comment]
Just a rebase to fix CI

**Pull request merged**